src/trace/dma-trace.c

```diff
@@ -475,26 +475,34 @@ void dma_trace_flush(void *t)
 void dma_trace_on(void)
 {
 	struct dma_trace_data *trace_data = dma_trace_data_get();
+	uint32_t flags;
 
 	if (!trace_data || trace_data->enabled)
 		return;
 
-	trace_data->enabled = 1;
+	spin_lock_irq(&trace_data->lock, flags);
```

Comment:
do we have a good understanding of what this lock is protecting, vs. what the other trace lock is protecting? The use of this lock doesn't seem to be fully consistent. E.g. `dtrace_event()` this lock is also covering the test of `.copy_in_progress`, but not in a consistent way, because both locations where `.copy_in_progress` is set to 1 aren't protected by that lock, so, holding the lock while testing it doesn't help. Why do we have to lock here at all?